src/main/drivers/accgyro/accgyro_spi_icm426xx.c

```diff
@@ -214,37 +228,33 @@ void icm426xxGyroInit(gyroDev_t *gyro)
     spiWriteReg(dev, ICM426XX_RA_ACCEL_CONFIG0, (3 - INV_FSR_16G) << 5 | (odrIdx & 0x0F));
     delay(15);
 
-    // Select desired AAF settings
-    uint8_t aafDelt = 0;
-    uint8_t aafBitshift = 0;
+    // Get desired settings for the gyro Anti-Alias Filter
+    aafConfig_t aafConfig;
     switch (gyroConfig()->gyro_hardware_lpf) {
-        case GYRO_HARDWARE_LPF_NORMAL:  // 258 Hz
-            aafDelt = ICM426XX_AAF_258HZ_DELT;
-            aafBitshift = ICM426XX_AAF_258HZ_BITSHIFT;
+        case GYRO_HARDWARE_LPF_NORMAL:
+            aafConfig = aafLUT[AAF_CONFIG_258HZ];
             break;
-        case GYRO_HARDWARE_LPF_OPTION_1:  // 536 Hz
-            aafDelt = ICM426XX_AAF_536HZ_DELT;
-            aafBitshift = ICM426XX_AAF_536HZ_BITSHIFT;
+        case GYRO_HARDWARE_LPF_OPTION_1:
+            aafConfig = aafLUT[AAF_CONFIG_536HZ];
             break;
-        case GYRO_HARDWARE_LPF_OPTION_2:  // 997 Hz
-            aafDelt = ICM426XX_AAF_997HZ_DELT;
-            aafBitshift = ICM426XX_AAF_997HZ_BITSHIFT;
+        case GYRO_HARDWARE_LPF_OPTION_2:
+            aafConfig = aafLUT[AAF_CONFIG_997HZ];
             break;
-        case GYRO_HARDWARE_LPF_EXPERIMENTAL:  // 1962 Hz
-            aafDelt = ICM426XX_AAF_1962HZ_DELT;
-            aafBitshift = ICM426XX_AAF_1962HZ_BITSHIFT;
+        case GYRO_HARDWARE_LPF_EXPERIMENTAL:
+            aafConfig = aafLUT[AAF_CONFIG_1962HZ];
             break;
```

Comment:
`default` is not handled here, with new GYRO_HARDWARE_LPF value, random values will be written into device. IMO gcc should warn about both - unhandled GYRO_HARDWARE_LPF_COUNT case and uninitialized aafConfig.
Also GYRO_HARDWARE_LPF_EXPERIMENTAL may be undefined. (https://github.com/betaflight/betaflight/blob/671f8abbc57d93fdceade12ab2fdbc8e3997ac2f/src/main/drivers/accgyro/accgyro.h#L70)